lib/install.js

```diff
@@ -95,69 +116,115 @@ install.completion = function (opts, cb) {
-var output = require('./utils/output.js')
-var saveMetrics = require('./utils/metrics.js').save
-
-// install specific libraries
-var audit = require('./install/audit.js')
-var {
-  getPrintFundingReport,
-  getPrintFundingReportJSON
-} = require('./install/fund.js')
-var errorMessage = require('./utils/error-message.js')
-
-const path = require('path')
-
-function install (where, args, cb) {
+async function install (where, args, cb) {
   if (!cb) {
     cb = args
     args = where
     where = null
   }
+
   // the /path/to/node_modules/..
   const globalTop = path.resolve(npm.globalDir, '..')
-  if (!where) {
-    where = npm.flatOptions.global
+  const {
+    dryRun,
+    global: isGlobalInstall
+  } = npm.flatOptions
+
+  // Determine where install is happening
+  // - If `where` is NOT set, then we need to determine if it's a global install
+  //    or if it's a local context install (hence nested ternary)
+  // - `where` IS SET, just use it
+  where = (!where)
+    ? (isGlobalInstall)
       ? globalTop
       : npm.prefix
-  }
-  const {dryRun} = npm.flatOptions
+    : where
 
   // TODO: Add warnings for other deprecated flags
   if (npm.config.get('dev')) {
     log.warn('install', 'Usage of the `--dev` option is deprecated. Use `--include=dev` instead.')
   }
 
+  // If global install, and NO args passed; user meant "this directory context"
   if (where === globalTop && !args.length) {
     args = ['.']
   }
+
+  // Filter our paths that are not THIS directory
   args = args.filter(a => path.resolve(a) !== npm.prefix)
 
-  const arb = new Arborist({
-    ...this.flatOptions,
-    path: where,
-  })
+  // Create arborist object to work with
+  const arb = new Arborist({ ...npm.flatOptions, path: where })
 
-  // TODO:
-  // - audit
-  // - funding
-  // - more logging (archy-ize the tree for silly logging)
-  // - global installs in Arborist
+  const arbCallOpts = { ...npm.flatOptions, add: args }
 
-  const opt = {
-    ...this.flatOptions,
-    add: args,
-  }
-  arb[dryRun ? 'buildIdealTree' : 'reify'](opt).then(tree => {
-    output('TREEEEEEEE', tree)
+  try {
+    const start = Date.now()
+
+    const tree = (dryRun)
+      ? await arb.buildIdealTree(arbCallOpts)
+      : await arb.reify(arbCallOpts)
+
+    const stop = Date.now()
+
+    // Print list of installed packages and their version
+    // Buffer with newline
+    output('')
+    arb.diff.children.forEach(diff => {
+      const node = diff.ideal
+      if (diff.action === 'ADD' && arb.explicitRequests.has(node.name)) {
+        output(`+ ${node.name}@${node.package.version}`)
+      }
+    })
+
+    /**
+     * TODO:
+     * - Add audit
+     * - Add funding
+     */
+
+    printResults({ start, stop }, tree, arb)
     cb()
-  }, er => cb(er))
+  } catch (err) {
+    const msg = errorMessage(err)
+    msg.summary.forEach(s => log.warn.apply(log, s))
+    msg.detail.forEach(d => log.verbose.apply(log, d))
+    cb(err)
+  }
+}
+
+function printResults (timing, tree, arb) {
```

Comment:
I think it probably would be more useful to silly-log a representation of the _diff_ rather than just the top-level deps.

Maybe just iterate through the nodes in the diff and report what we're doing at each path (adding, removing, or replacing) and the version/integrity of what's getting put there?